modules/metastore/src/Service.php

```diff
@@ -137,54 +139,33 @@ public function count(string $schema_id): int {
    *   Schema object offset or null for all.
    * @param int|null $length
    *   Number of objects to fetch or null for all.
+   * @param bool $unpublished
+   *   Whether to include unpublished items.
    *
    * @return string[]
    *   Range of object UUIDs of the given schema_id.
    */
-  public function getRangeUuids(string $schema_id, ?int $start = NULL, ?int $length = NULL): array {
-    return $this->getStorage($schema_id)->retrieveRangeUuids($start, $length);
+  public function getIdentifiers(string $schema_id, ?int $start = NULL, ?int $length = NULL, $unpublished = FALSE): array {
+    return $this->getStorage($schema_id)->retrieveIds($start, $length, $unpublished);
   }
 
   /**
-   * Get all.
-   *
-   * @param string $schema_id
-   *   The {schema_id} slug from the HTTP request.
-   *
-   * @return array
-   *   All objects of the given schema_id.
-   */
-  public function getAll($schema_id): array {
-    $jsonStringsArray = $this->getStorage($schema_id)->retrieveAll();
-    $objects = array_filter($this->jsonStringsArrayToObjects($jsonStringsArray, $schema_id));
-
-    return $this->dispatchEvent(self::EVENT_DATA_GET_ALL, $objects, function ($data) {
-      if (!is_array($data)) {
-        return FALSE;
-      }
-      if (count($data) == 0) {
-        return TRUE;
-      }
-      return reset($data) instanceof RootedJsonData;
-    });
-
-  }
-
-  /**
-   * Get a subset of metastore items according to a range.
+   * Get all items, with optional pagination and status filters.
    *
    * @param string $schema_id
    *   Schema ID.
-   * @param int $start
-   *   Start offset.
-   * @param int $length
+   * @param int|null $start
+   *   Start offset. Null for no range.
+   * @param int|null $length
    *   Number of items to retrieve.
+   * @param bool $unpublished
+   *   Whether to include unpublished items. Should default to false.
    *
    * @return array
    *   Array of RootedJsonData objects.
    */
-  public function getRange(string $schema_id, int $start, int $length):array {
-    $jsonStringsArray = $this->getStorage($schema_id)->retrieveRange($start, $length);
+  public function getAll(string $schema_id, ?int $start = NULL, ?int $length = NULL, $unpublished = FALSE):array {
```

Comment:
Slight nitpick: `getAll()` sounds weird to me since this no longer gets all of the metastore items for the given schema. How does `getMultiple()` sound?

Reply:
I had the same thought, but ultimately decided to punt on it for similar reasons I decided against removing that interface. I'd like to think more carefully and in a group about the best naming for all these methods, and also I think public methods on a service should be the ones we want to be the most cautious about renaming or removing.

Think about it this way: we are still getting "all" the datasets. We may want to paginate the results in certain contexts for performance/scaling reasons, and "all" is ambiguous because it could mean all the public datasets or all the datasets including unpublished, but there is no further filtering going on here.

I'd propose we keep it getAll at least for this week, and have that conversation or conversations about further metastore refactoring next week and just jump to 2.14 if we decide renaming getAll is still the best course of action.

Reply:
Ok, fair enough. This would be considered a BC-breaking change if someone were making use of the metastore service in some custom code. I agree that we need to have a larger conversation around refactoring efforts.